pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -278,17 +289,38 @@ func (c *DeploymentConfigController) reconcileDeployments(existingDeployments *k
 }
 
 func (c *DeploymentConfigController) updateStatus(config *deployapi.DeploymentConfig) error {
-	if config.Generation > config.Status.ObservedGeneration {
-		config.Status.ObservedGeneration = config.Generation
+	// NOTE: We should update the status of the deployment config only if we need to, otherwise
+	// we hotloop between updates.
+	if !needsUpdate(config) {
+		return nil
 	}
-	if _, err := c.osClient.DeploymentConfigs(config.Namespace).UpdateStatus(config); err != nil {
+	config.Status.ObservedGeneration = config.Generation
+	if _, err := c.dn.DeploymentConfigs(config.Namespace).UpdateStatus(config); err != nil {
 		glog.V(2).Infof("Cannot update the status for %q: %v", deployutil.LabelForDeploymentConfig(config), err)
-		return transientError(fmt.Sprintf("cannot update the status for %q - requeuing", deployutil.LabelForDeploymentConfig(config)))
+		return transientError(err.Error())
 	}
 	glog.V(4).Infof("Updated the status for %q (observed generation: %d)", deployutil.LabelForDeploymentConfig(config), config.Status.ObservedGeneration)
 	return nil
 }
 
+func (c *DeploymentConfigController) handleErr(err error, key interface{}) {
+	if _, isFatal := err.(fatalError); isFatal {
+		utilruntime.HandleError(err)
+		c.queue.Forget(key)
+		return
+	}
+	// requeue deployment configs with transient errors.
+	if _, isTransient := err.(transientError); isTransient {
+		c.queue.AddRateLimited(key)
+		return
+	}
+	utilruntime.HandleError(err)
```

Comment:
So by default, errors that are neither fatal, nor transient are treated as fatal?  You sure you want to default in that direction?  I'm ok with it, but sometimes you'll be getting APIs errors (update errors as a for instance), that are likely to be worth a retry.